packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -2881,53 +2880,32 @@ describe('<Autocomplete />', () => {
     });
   });
 
-  it('should not override internal listbox ref when external listbox ref is provided', () => {
+  it('should not override internal listbox ref when external listbox ref is provided by testing if highlighting works', () => {
     const handleHighlightChange = spy();
-    const options = ['one', 'two', 'three'];
-    const ref = React.createRef(null);
+    const externalListboxRef = React.createRef(null);
+
     render(
       <Autocomplete
-        defaultValue={options[0]}
-        onHighlightChange={handleHighlightChange}
-        options={options}
-        ListboxProps={{ ref }}
-        open
+        options={['one', 'two', 'three']}
+        slotProps={{
+          listbox: {
+            ref: externalListboxRef,
+          },
+        }}
         renderInput={(params) => <TextField {...params} autoFocus />}
+        onHighlightChange={handleHighlightChange}
       />,
     );
-    expect(handleHighlightChange.callCount).to.equal(
-      // FIXME: highlighted index implementation should be implemented using React not the DOM.
-      reactMajor > 18 ? 2 : 1,
-    );
-    expect(handleHighlightChange.args[0]).to.deep.equal([undefined, options[0], 'auto']);
-    if (reactMajor > 18) {
-      expect(handleHighlightChange.args[1]).to.deep.equal([undefined, options[0], 'auto']);
-    }
+
+    const textbox = screen.getByRole('combobox');
+
+    fireEvent.keyDown(textbox, { key: 'ArrowDown' }); // open listbox
+    fireEvent.keyDown(textbox, { key: 'ArrowDown' }); // highlight first option
+
+    expect(handleHighlightChange.callCount).to.equal(1);
   });
 
   describe('prop: onHighlightChange', () => {
-    it('should trigger event when default value is passed', () => {
```

Comment:
Should we inverse this test? i.e. make it validate no event is triggered when the default value is passed.

Reply:
We could. Done.